front/components/dialogs/shareIdeaByEmail.vue

```diff
@@ -83,7 +90,21 @@ export default {
         return
       }
 
-      this.$emit('success')
+      // Call shre idea over email api.
+
+      this.closeDialog()
+    },
+    closeDialog() {
+      this.$emit('close')
+
+      setTimeout(() => {
```

Comment:
why timeout? Why not requestAnimationFrame without timeout ?

Reply:
I haven't used it before, can you provide one example?